### PR TITLE
Fix(BWS) - opts.from gets set by BWS instead of Copay

### PR DESCRIPTION
**Pull request by justinkook**
fix address validation for ETH
throw Address Errors on validation

### CODE REVIEW

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 1]
```diff
@@ -2608,6 +2612,13 @@ export class WalletService {
 
             async.series(
               [
+                (next) => {
+                  this.getMainAddresses({reverse: true, limit: 1}, (err, mainAddr) => {
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/server.ts]: lets do this only on the chain that need it. Many add `needsFromAddress` flag in chain Service or something?

File: packages/bitcore-wallet-service/src/lib/server.ts [hunk 2]
```diff
@@ -2360,8 +2360,12 @@ export class WalletService {
       const output = opts.outputs[i];
       output.valid = false;
 
-      const addrErr = ChainService.validateAddress(wallet, output.toAddress, opts);
-      if (addrErr) return addrErr;
+      try {
+        ChainService.validateAddress(wallet, output.toAddress, opts);
+        ChainService.validateAddress(wallet, opts.from, opts);
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/server.ts]: only on certain chains

Comment (justinkook) [reply packages/bitcore-wallet-service/src/lib/server.ts]: fixed to use opts third param for ETH/XRP only

File: packages/bitcore-wallet-service/src/lib/server.ts (hunk 1 above)
Comment (justinkook) [reply packages/bitcore-wallet-service/src/lib/server.ts]: added `isUTXOCoin` check

**Pull request merged**